### PR TITLE
Taglib 1.11.1 support

### LINKED ISSUE #77: Setting a Playlist ID to an M4A file does not work properly.
Hi!

First off, I want to say thanks for this great wrapper for TagLib! However, there is a problem when trying to set a playlist ID to a m4a file. If I execute the code below:

```ruby
TagLib::MP4::File.open("song.m4a") do |tune|
  item = TagLib::MP4::Item.from_int(123456789)
  tune.tag.item_list_map.insert('plID', item)
  tune.save
end
```

The playlist id then gets assigned a long random number which is not the number supplied (In this case I checked using mp4info which yielded 72057594161384725 for the playlist id, using a random sample m4a).

Its quite strange as it works on occasion, but more times than not it fails and stores a very large, random number for the song. Any help would be appreciated.

Thanks.

### LINKED ISSUE #78: Support for Taglib 1.11.1
Hi everyone,

Taglib 1.11.1 starts being quite mainstream: Debian stable, Ubuntu 17.10, Fedora 25/26/27, the next Mint...

Unfortunately, Taglib 1.11 introduced a new way to handle pictures in Ogg files. They were previously accessed as standard tags (base64-encoded), but are now available as a list of FLAC::Picture objects. Which means that on those modern distributions, Taglib-ruby (which is based on Taglib 1.7 if I'm correct) can't access pictures from Ogg files anymore.

Since a tool of mine requires this feature, I'm willing to try to upgrade Taglib-ruby for a Taglib 1.11.1 support. I've never used Swig or native gems before, just played a bit on the last few weeks, but it seems doable and fun.

Any thought/advice/suggestion?

### PR DESCRIPTION
**Pull request by Sigill**
TagLib 1.11 introduced a new API to access and edit pictures of Ogg files (preventing pictures from being accessed using the old API).
Among other few things, this patch bring support for this new API, dropping support for older versions of TagLib (closes #78).

Methods flagged as deprecated in TagLib-1.11.1 have been removed.
Some APIs have been purposely ignored:
- the new "Unified property interface" API, which allow to edit some file types by 1/exporting the properties as a map, 2/ editing the map, 3/ reimporting the whole map.
- various TagLib::*::File constructors reading from a C++ IOStream
- APIs that appear to only be useful internally

Unit-tests have been updated/added for most of the new stuff.
Yard documentation have been updated (assuming that the new release will be numbered 1.0.0, as discussed in #80).

The cxx files have been generated using Swig-3.0.7. Version 3.0.10, which is currently the default on Debian stable, has a memory management issue in Ruby wrappers (release 3.0.8 to 3.0.12 are concerned, a fix has been integrated).

This should also close #77.

### CODE REVIEW

File: lib/taglib/ogg.rb [hunk 1]
```diff
@@ -1 +1,14 @@
...
+module TagLib::Ogg
+  class XiphComment
+    # alias :field_list_map :_field_list_map
+    # remove_method :_field_list_map
+
+    # alias :add_field :_add_field
+    # remove_method :_add_field
+
+    # alias :remove_field :_remove_field
+    # remove_method :_remove_field
```
Comment (robinst) [inline lib/taglib/ogg.rb]: Remove these?

File: README.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ Contributions for more coverage of the library are very welcome.
 Installation
 ------------
 
-Before you install the gem, make sure to have [taglib][taglib] installed
+Before you install the gem, make sure to have [taglib 1.11.1][taglib] installed
```
Comment (robinst) [inline README.md]: Change this to be "1.11.1 or higher"?

**Comment by robinst** [comment]
Regarding the IOStream methods, there was a request a while back of being able to read in-memory data, see #12. I'm not sure how easy it would be to wrap a Ruby IO as an IOStream, but if that's too hard we could probably provide functionality for reading from a Ruby String. Anyway, we can look at that separately from this PR.

File: lib/taglib/ogg.rb (hunk 1 above)
Comment (Sigill) [reply lib/taglib/ogg.rb]: Removed

File: README.md (hunk 2 above)
Comment (Sigill) [reply README.md]: Changed

**Comment by Sigill** [comment]
I took a quick look at the IOStream classes and I think there's a bit of work to make them usable from the Ruby API. I'll leave that to someone else, I probably won't have time to work on that.

**Comment by robinst** [comment]
Merged!

I've also given you push access to this repository, you can now help maintain it if you want. As you can probably tell, @jacobvosmaer and me haven't been very active in this project lately :).

Do you want to do the release of 1.0.0? I could give you access to publish the gem as well. We need to add a changelog entry (in CHANGES.md) first though.

**Comment by Sigill** [comment]
I'll do my best to help you!
Let's do a release. Do you want me to push the changelog and version bump?

**Comment by robinst** [comment]
Cool! Yes that would be awesome. I'm currently on holidays so I might be a bit slow with replying :).

Yeah, create a pull request with the changelog and version bump. Oh I also have to figure out gem publishing access.

**Comment by robinst** [comment]
I've added you to the owners now, alongside me and @jacobvosmaer: https://rubygems.org/gems/taglib-ruby

So after we've done the changelog, we can look at the release.

**Comment by chrsgrrtt** [comment]
Just wondered if this release ever made it out? We're using 0.7.1 from rubygems.org but seem to be missing methods like `bits_per_sample` that are on [master](https://github.com/robinst/taglib-ruby/blob/fc49ec908c38aeff93d231f2b92fb9e85c85f097/test/flac_file_test.rb#L55).

**Comment by jacobvosmaer** [comment]
The release got stalled in https://github.com/robinst/taglib-ruby/pull/84

**Comment by Sigill** [comment]
Yeah, sorry, I never took the time to finish #84. I'll try to do something this week-end.

In the meantime, I discovered that release 1.11.1 of Taglib is broken (https://github.com/taglib/taglib/issues/864, which is fixed but has never been released), and it does not seems there's plan to do new releases.

**Pull request merged**